src/hotspot/share/prims/jvmtiEnvBase.cpp

```diff
@@ -1549,16 +1549,21 @@ JvmtiEnvBase::get_object_monitor_usage(JavaThread* calling_thread, jobject objec
             // it has not been notified. This thread can't change the
             // state of the monitor so it doesn't need to be suspended.
             Handle th(current_thread, get_vthread_or_thread_oop(w));
-            ret.waiters[offset + j] = (jthread)jni_reference(calling_thread, th);
-            ret.notify_waiters[j++] = (jthread)jni_reference(calling_thread, th);
+            if (java_lang_Thread::get_thread_status(w->threadObj()) ==
+                JavaThreadStatus::BLOCKED_ON_MONITOR_ENTER) {
```

Comment:
I don't think this is possible. `BLOCKED_ON_MONITOR_ENTER` is only set after the target has been removed from the wait-set and added to the cxq queue - see `ObjectMonitor::INotify`. As per the comment just above:
```
// If the thread was found on the ObjectWaiter list, then
// it has not been notified.
```
which means it can't be trying to acquire the monitor either.

Reply:
Thank you for looking at it and the comment, David! Will check what is possible here.
Do you have any suggestions?